apps/prairielearn/src/lib/groups.sql

```diff
@@ -261,3 +261,60 @@ VALUES
     $group_id,
     'update roles'
   );
+
+-- BLOCK delete_all_groups
+WITH
+  assessment_groups AS (
+        SELECT
+          g.id
+        FROM
+          group_configs AS gc
+          JOIN groups AS g ON (g.group_config_id = gc.id)
+        WHERE
+          gc.assessment_id = $assessment_id
+          AND g.deleted_at IS NULL
+          AND gc.deleted_at IS NULL
+  ),
+  deleted_group_users AS (
+    DELETE FROM group_users
+    WHERE
+      group_id IN (
+        SELECT
+          id
+        FROM
+          assessment_groups
+      )
+    RETURNING
+      user_id,
+      group_id
+  ),
+  deleted_group_users_logs AS (
+    INSERT INTO
+      group_logs (authn_user_id, user_id, group_id, action)
+    SELECT
+      $authn_user_id,
+      user_id,
+      group_id,
+      'leave'
+    FROM
+      deleted_group_users
+  ),
+  deleted_groups AS (
+    UPDATE groups AS g
+    SET
+      deleted_at = NOW()
+    FROM
+      assessment_groups AS ag
+    WHERE
+      g.id = ag.id
+    RETURNING
+      id
```

Comment:
```suggestion
      g.id
```

This gave me an error when testing locally.